src/pages/Auth/Signup/SignupPage/components/IdInput.tsx

```diff
@@ -1,13 +1,43 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import cn from 'utils/ts/classNames';
 import { ReactComponent as ErrorIcon } from 'assets/svg/auth/error.svg';
 import { useFormContext } from 'react-hook-form';
+import { useQuery } from 'react-query';
+import { checkIdDuplicate } from 'api/user';
 import { ERROR_MESSAGE } from '../../static/signUp';
 import styles from '../SignUp.module.scss';
 import { SignUpFormData } from '../entity';
 
+const useIdCheckServer = (id: string) => {
+  const { status } = useQuery(['idDuplicate', id], () => checkIdDuplicate({ account: id }), {
```

Comment:
id를 굳이 밖에서 가져올 필요 없이 파라미터 첫번째인 `context`에서 가져오면 됩니다.
```suggestion
  const { status } = useQuery(['idDuplicate', id], ([_,account]) => checkIdDuplicate({ account }), {
```
위와 같이 되겠네요.

Reply:
queryFn 콜백함수의 인자로 쿼리키를 갖고오는군요! 👍

Reply:
적용하려고 하니 타입이 맞지 않아 문서 타입을 참고했는데, 쿼리에 대한 정보를 객체로 받아 `queryKey`라는 key로 접근할 수 있습니다. 
아래처럼 적용해야 할 것 같은데, 어떻게 생각하시나요?
```tsx
const { status } = useQuery(['idDuplicate', id], ({ queryKey }) => checkIdDuplicate({ account: queryKey[1] }), {
```

Reply:
```suggestion
  const { status } = useQuery(['idDuplicate', id], ({queryKey: [_, account]}) => checkIdDuplicate({ account }), {
```
로 바꿀 수 있겠네요. Context를 갖고온다는 사실을 까먹고 있었네요.

Reply:
수정했습니다~